src/core/client/stream/mutations/CreateCommentMutation.ts

```diff
@@ -5,144 +5,69 @@ import {
   RecordSourceSelectorProxy,
 } from "relay-runtime";
 
-import { getMe, getMeSourceID } from "talk-framework/helpers";
+import { getMe } from "talk-framework/helpers";
 import { TalkContext } from "talk-framework/lib/bootstrap";
 import {
   commitMutationPromiseNormalized,
   createMutationContainer,
 } from "talk-framework/lib/relay";
 import { Omit } from "talk-framework/types";
-
 import { CreateCommentMutation as MutationTypes } from "talk-stream/__generated__/CreateCommentMutation.graphql";
 
+import {
+  incrementStoryCommentCounts,
+  prependCommentEdgeToProfile,
+} from "../helpers";
+
 export type CreateCommentInput = Omit<
   MutationTypes["variables"]["input"],
   "clientMutationId"
-> & { local?: boolean };
+>;
 
 function sharedUpdater(
   environment: Environment,
   store: RecordSourceSelectorProxy,
   input: CreateCommentInput
 ) {
-  updateStory(store, input);
-  if (input.local) {
-    localUpdate(store, input);
-  } else {
-    update(store, input);
-  }
-  updateProfile(environment, store, input);
-}
-
-function updateStory(
-  store: RecordSourceSelectorProxy,
-  input: CreateCommentInput
-) {
-  // Updating Comment Count
-  const story = store.get(input.storyID);
-  if (story) {
-    const record = story.getLinkedRecord("commentCounts");
-    if (record) {
-      // TODO: when we have moderation, we'll need to be careful here.
-      const currentCount = record.getValue("totalVisible");
-      record.setValue(currentCount + 1, "totalVisible");
-    }
-  }
+  incrementStoryCommentCounts(store, input.storyID);
+  prependCommentEdgeToProfile(
+    environment,
+    store,
+    store.getRootField("createComment")!.getLinkedRecord("edge")!
+  );
+  addCommentToStory(store, input);
 }
 
 /**
  * update integrates new comment into the CommentConnection.
  */
-function update(store: RecordSourceSelectorProxy, input: CreateCommentInput) {
+function addCommentToStory(
+  store: RecordSourceSelectorProxy,
+  input: CreateCommentInput
+) {
   // Get the payload returned from the server.
   const payload = store.getRootField("createComment")!;
 
   // Get the edge of the newly created comment.
   const newEdge = payload.getLinkedRecord("edge")!;
 
-  // Get parent proxy.
-  const parentProxy = input.parentID
-    ? store.get(input.parentID)
-    : store.get(input.storyID);
-
-  const connectionKey = input.parentID
-    ? "ReplyList_replies"
-    : "Stream_comments";
+  // Get stream proxy.
+  const streamProxy = store.get(input.storyID);
+  const connectionKey = "Stream_comments";
+  const filters = { orderBy: "CREATED_AT_DESC" };
```

Comment:
Maybe you ca replace these with values from the GraphQL schema?

`GQLCOMMENT_SORT.CREATED_AT_DESC`?